src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -566,33 +634,27 @@ public ByteStringContext<ByteStringMemoryCache>.InternalScope Finish(out ByteStr
 
         // The size of the known fields metadata section
         int metadataSection = IndexEntryReader.TableEncodingLookupTable[(int)encodeSize] * _knownFields.Count;
-        // The size of the unknown/dynamic fields metadata section            
-        int dynamicMetadataSection = _dynamicFieldIndex * sizeof(uint);
-        int dynamicMetadataSectionOffset = buffer.Length - dynamicMetadataSection - 1;
 
         ref var header = ref MemoryMarshal.AsRef<IndexEntryHeader>(buffer);
-        header.Length = (uint)(_dataIndex + dynamicMetadataSection + metadataSection + 1);
 
         // The known field count is encoded as xxxxxxyy where:
-        // x: the count
+        // x: the count (1 ... 16K should be enough, using checked math to verify it anyway)
         // y: the encode size
-        header.KnownFieldCount = (ushort)(_knownFields.Count << 2 | (int)encodeSize);
-        header.DynamicTable = (uint)_dataIndex;
+        header.KnownFieldCount = checked((ushort)(_knownFields.Count << 2 | (int)encodeSize));
 
         // The dynamic metadata fields count. 
-        Unsafe.WriteUnaligned(ref buffer[_dataIndex], (byte)_dynamicFieldIndex);
-        _dataIndex += sizeof(byte);
-
-        if (dynamicMetadataSection != 0)
+        header.DynamicTable = (uint)_dataIndex;
+        _dataIndex += VariableSizeEncoding.Write(buffer, numberOfDynamicFields, _dataIndex);
+        if (_dynamicFieldsLocations != null)
         {
-            // From the offset to the end... move the data toward the closest position
-            var metadataTable = buffer[dynamicMetadataSectionOffset..];
-            metadataTable.CopyTo(buffer[_dataIndex..]);
-
-            // Move the pointer to the end of the copied section.
-            _dataIndex += dynamicMetadataSection;
+            foreach (int fieldsLocation in _dynamicFieldsLocations!)
```

Comment:
Dynamics fields pointers MUST be sorted by name.

Reply:
Why? 
We assume that there are going to be relatively few of them.
Likely 1 - 2 per entry.

We can just do linear scan. It is _highly_ unlikely that you'll need more.

Also remember what we'll need to use that for.
Either to list all the entries during indexing  - scan is sufficent.
Or if you sort by dynamic field (pretty rare).

Reply:
Not really, to access them we need to scan and do field name comparison. And we dont know how many dynamic fields we have on the index itself. That worst case scenario of the 100 fields in a single index is still probable. Changing that would involve very low level work to be done that is error prone, not an easy fix. I would do it correctly before we have to deal with that in production.

Reply:
Not really how that works. This is at the level of the _index entry_, not the global index.
And the only scenario where we actually need to do a lookup by name (vs. a scan) is for sorting by dynamic fields.

Reply:
Forgot about that. We can do the search faster with SIMD instruction if that is the case. I would still sort them though.